helper/scaleutils/filter.go

```diff
@@ -0,0 +1,99 @@
+package scaleutils
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/nomad/api"
+)
+
+// NodeIDMap provides a mapping between the Nomad ID of a node and its remote
+// infrastructure provider specific ID.
+type NodeIDMap struct {
```

Comment:
`NodeIDMap` sounds like it would be a `map`, which is not the case. Maybe just `NodeID` since it would scoped as `scaleutils.NodeID` outside the package.